dlib/cuda/cuda_dlib.h

```diff
@@ -464,13 +520,13 @@ namespace dlib
                     memcpy(buf + i*bytes_per_plane, &t(0,0), bytes_per_plane);
                 }
 
-                do_work(static_cast<float*>(loss_buf.data()), static_cast<uint16_t*>(buf.data()), subnetwork_output, gradient, loss);
+                do_work(loss_buf, static_cast<uint16_t*>(buf.data()), subnetwork_output, gradient, loss);
             }
 
         private:
 
             static void do_work(
-                float* loss_cuda_work_buffer,
+                cuda_data_ptr<float> loss_cuda_work_buffer,
```

Comment:
Same comments here about cuda naming and `truth_buffer` (i.e. it should be a cuda_data_ptr too so the arguments are consistent)